api/server/server.go

```diff
@@ -1340,34 +1338,29 @@ func postContainerExecCreate(eng *engine.Engine, version version.Version, w http
 		return nil
 	}
 	var (
-		name         = vars["name"]
-		job          = eng.Job("execCreate", name)
-		stdoutBuffer = bytes.NewBuffer(nil)
-		outWarnings  []string
-		warnings     = bytes.NewBuffer(nil)
+		name = vars["name"]
 	)
 
-	if err := job.DecodeEnv(r.Body); err != nil {
+	execConfig := runconfig.ExecConfig{}
```

Comment:
As far as I know, the execConfig will only be used in `processConfig := execdriver.ProcessConfig`[https://github.com/docker/docker/blob/master/daemon/exec.go#L138]. And any fields that didn't decode in the old code will also not be used in the new code. By the way, I think two commits is OK, as they are done step by step. If you insist that I should squash, I'll do that tomorrow. Thanks for your review.  :)